core/src/main/java/org/apache/accumulo/core/summary/SummaryCollection.java

```diff
@@ -0,0 +1,167 @@
+    long filesExceedingBoundry;
+    long filesLarge;
+
+    public MergedSummary(FileSummary entry) {
+      this.summary = entry.summary;
+      this.filesContaining = 1;
+      this.filesExceedingBoundry = entry.exceededBoundry ? 1 : 0;
+      this.filesLarge = entry.exceededMaxSize ? 1 : 0;
+    }
+
+    public MergedSummary(TSummary tSummary) {
+      this.summary = new HashMap<>(tSummary.getSummary());
+      this.filesContaining = tSummary.getFilesContaining();
+      this.filesExceedingBoundry = tSummary.getFilesExceeding();
+      this.filesLarge = tSummary.getFilesLarge();
+    }
+
+    public void merge(MergedSummary other, SummarizerConfiguration config, SummarizerFactory factory) {
+
+      if (summary == null && other.summary != null) {
+        summary = new HashMap<>(other.summary);
+      } else if (summary != null && other.summary != null) {
+        Summarizer summarizer = factory.getSummarizer(config);
+        summarizer.combiner(config).merge(summary, other.summary);
+      }
+
+      filesContaining += other.filesContaining;
+      filesExceedingBoundry += other.filesExceedingBoundry;
+      filesLarge += other.filesLarge;
+    }
+
+    public TSummary toThrift(SummarizerConfiguration key) {
+      TSummarizerConfiguration tsumConf = SummarizerConfigurationUtil.toThrift(key);
+      return new TSummary(summary, tsumConf, filesContaining, filesExceedingBoundry, filesLarge);
+    }
+
+  }
+
+  private Map<SummarizerConfiguration,MergedSummary> mergedSummaries;
+  private long totalFiles;
+
+  public SummaryCollection() {
+    mergedSummaries = new HashMap<>();
+    totalFiles = 0;
+  }
+
+  public SummaryCollection(TSummaries tsums) {
+    mergedSummaries = new HashMap<>();
+    for (TSummary tSummary : tsums.getSummaries()) {
+      SummarizerConfiguration sconf = SummarizerConfigurationUtil.fromThrift(tSummary.getConfig());
+      mergedSummaries.put(sconf, new MergedSummary(tSummary));
+    }
+
+    totalFiles = tsums.getTotalFiles();
+  }
+
+  SummaryCollection(Collection<FileSummary> initialEntries) {
+    mergedSummaries = new HashMap<>();
+    for (FileSummary entry : initialEntries) {
+      mergedSummaries.put(entry.conf, new MergedSummary(entry));
+    }
+    totalFiles = 1;
+  }
+
+  static class FileSummary {
+
+    private SummarizerConfiguration conf;
+    private Map<String,Long> summary;
+    private boolean exceededBoundry;
+    private boolean exceededMaxSize;
+
+    FileSummary(SummarizerConfiguration conf, Map<String,Long> summary, boolean exceededBoundry) {
+      this.conf = conf;
+      this.summary = summary;
+      this.exceededBoundry = exceededBoundry;
+      this.exceededMaxSize = false;
+    }
+
+    FileSummary(SummarizerConfiguration conf) {
+      this.conf = conf;
+      this.summary = new HashMap<>();
+      ;
+      this.exceededBoundry = false;
+      this.exceededMaxSize = true;
+    }
+  }
+
+  public void merge(SummaryCollection other, SummarizerFactory factory) {
```

Comment:
I don't think so.  Alot of code is assumed to be single threaded.  Also, its internal as you mentioned.